chain/src/chain.rs

```diff
@@ -791,11 +791,11 @@ impl Chain {
 		let mut oldest_height = 0;
 		let mut oldest_hash = ZERO_HASH;
 
-		let mut current = self.get_block_header(&header_head.last_block_h);
```

Comment:
I don't think this change makes sense.
We are checking to see if a `header` exists in the db and we should do this based on the current `header_head` which is the tip of the current header chain.

We are in `check_txhashset_needed` which (confusingly) returns a vec of block hashes to go request (if we do not need to request a txhashset archive).
So we need to determine which blocks we do not have, based on the headers we do have. So we should be looking at the headers in both our db and our header MMR. Headers up to and including the current tip of the header chain (`header_head`).